handlers/Base/index.js

```diff
@@ -3,21 +3,51 @@
 
 require('./styles.css');
 
-import React from 'react';
+import React, {Component, PropTypes} from 'react';
 import {RouteHandler} from 'react-router';
 import Navigation from 'Navigation';
 import Footer from 'Footer';
-import Hamburger from 'Hamburger';
 
-class AppBase extends React.Component {
-  constructor(props, context) {
+class AppBase extends Component {
+  constructor(props: mixed, context: mixed): void {
     super(props, context);
 
     this.state = {navVisible: false};
     this.toggleNav = this.toggleNav.bind(this);
+    this.topOfViewport = true;
   }
 
-  getHandlerKey() {
+  componentDidMount(): void {
+    if (typeof window === 'undefined') return;
+    window.addEventListener('scroll', _ => {
```

Comment:
since this is the base component it is never unmounted, therefore my unsafe, never remove event listener.